controller/Dockerfile

```diff
@@ -7,4 +7,10 @@ ADD controller /controller
 # just in case
 RUN chmod a+x /controller
 
+# Production image
+FROM gcr.io/distroless/static:latest
+COPY --from=intermediate /controller /controller
+
+USER 65534
```

Comment:
Any reason to use the `USER` directive instead of using the k8s pod spec?

Reply:
I wanted to clearly indicate that the image runs as non-root which is the default behaviour. On the other hand, the gain here is that you do not specify the `runAsUser` to actually run as `65534` - I know it is already specified in the k8s manifest.

Reply:
Well, you can have a PSP in your cluster that mutates the pod spec to not run as root too. So, no need to specify it manually on the pod spec.

I think this is a different concern and I'd leave it for another PR. I think I'd lean towards not specifying the `USER` in the docker image, in fact. I think specifying that on Kubernetes makes more sense (there is little we can know about which user to use here in the MetalLB Dockerfile to choose the user), it can be done automatically via PSPs (no need to manually modify the pod spec) and if the container runtime support user namespaces (we at Kinvolk are working to add that to Kubernetes upstream) it has little sense to have some default at the Dockerfile layer, I think.

In any case, we can discuss and see what is best for the `USER` in another issue/PR and just keep this for distroless. What do you think?

Thanks!

Reply:
Agreed 👌 will rewrite the commit without `USER`